diff.js

```diff
@@ -73,6 +74,9 @@ function diff_main(text1, text2) {
     diffs.push([DIFF_EQUAL, commonsuffix]);
   }
   diff_cleanupMerge(diffs);
+  if (cursor_pos != null) {
```

Comment:
`diff_cleanupMerge` also shifts edits (see line 541), reversing some of my modifications:

```
diff_cleanupMerge(fix_cursor([[0, 'aa'], [1, 'a']], 1)) 
= diff_cleanupMerge([[0, 'a'], [1, 'a'], [0, 'a']]) 
= [[0, 'aa'], [1, 'a']]
```

I could extend `diff_cleanupMerge` to accept some flag to ignore the shifting part. But then there is also the problem that it checks the whole array, and also performs other modifications I don't fully understand.